src/vs/workbench/contrib/output/browser/output.contribution.ts

```diff
@@ -334,6 +337,75 @@ class OutputContribution extends Disposable implements IWorkbenchContribution {
 		return null;
 	}
 
+	private registerConfigureActiveOutputLogLevelAction(): void {
+		const that = this;
+		const logLevelMenu = new MenuId('workbench.output.menu.logLevel');
+		this._register(MenuRegistry.appendMenuItem(MenuId.ViewTitle, {
+			submenu: logLevelMenu,
+			title: nls.localize('logLevel.label', "Set Log Level..."),
+			group: 'navigation',
+			when: ContextKeyExpr.and(ContextKeyExpr.equals('view', OUTPUT_VIEW_ID), CONTEXT_ACTIVE_OUTPUT_LEVEL_SETTABLE),
+			icon: Codicon.gear,
+			order: 6
+		}));
+		const registeredLogLevels = new Map<LogLevel, IDisposable>();
+		this._register(toDisposable(() => dispose(registeredLogLevels.values())));
+		const registerLogLevels = (logLevels: LogLevel[]) => {
+			let order = 0;
+			for (const logLevel of logLevels) {
+				const { original, value } = LogLevelToLocalizedString(logLevel);
+				registeredLogLevels.set(logLevel, registerAction2(class extends Action2 {
+					constructor() {
+						super({
+							id: `workbench.action.output.activeOutputLogLevel.${logLevel}`,
+							title: value,
+							toggled: CONTEXT_ACTIVE_OUTPUT_LEVEL.isEqualTo(original),
+							menu: {
+								id: logLevelMenu,
+								order: order++,
+								group: '0_level'
+							}
+						});
+					}
+					async run(accessor: ServicesAccessor): Promise<void> {
+						const channel = that.outputService.getActiveChannel();
+						if (channel) {
+							const channelDescriptor = that.outputService.getChannelDescriptor(channel.id);
+							if (channelDescriptor?.log && channelDescriptor.file) {
+								return accessor.get(ILoggerService).setLogLevel(channelDescriptor.file, logLevel);
+							}
+						}
+					}
+				}));
+			}
+			this._register(registerAction2(class extends Action2 {
+				constructor() {
+					super({
+						id: `workbench.action.output.activeOutputLogLevelDefault`,
+						title: nls.localize('logLevelDefault.label', "Set As Default"),
+						menu: {
+							id: logLevelMenu,
+							order,
+							group: '1_default'
+						},
+						precondition: CONTEXT_ACTIVE_OUTPUT_LEVEL_IS_DEFAULT.negate()
+					});
+				}
+				async run(accessor: ServicesAccessor): Promise<void> {
+					const channel = that.outputService.getActiveChannel();
+					if (channel) {
+						const channelDescriptor = that.outputService.getChannelDescriptor(channel.id);
+						if (channelDescriptor?.log && channelDescriptor.file) {
+							const logLevel = accessor.get(ILoggerService).getLogLevel(channelDescriptor.file);
+							return await accessor.get(IDefaultLogLevelsService).setDefaultLogLevel(logLevel, channelDescriptor.extensionId);
+						}
+					}
+				}
+			}));
+		};
+		registerLogLevels([LogLevel.Trace, LogLevel.Debug, LogLevel.Info, LogLevel.Warning, LogLevel.Error, LogLevel.Off]);
```

Comment:
Its personal preference: I would prefer to register action on each log level instead of registering the action in a for loop. It is easier to understand.

Reply:
I have pushed a change to do this.